source/common/ssl/context_config_impl.cc

```diff
@@ -62,6 +68,27 @@ ContextConfigImpl::ContextConfigImpl(const envoy::api::v2::CommonTlsContext& con
   }
 }
 
+unsigned
+ContextConfigImpl::tlsVersionFromProto(const envoy::api::v2::TlsParameters_TlsProtocol& version,
```

Comment:
It would probably not be a bad idea to have a small unit test that this function works as expected with auto, etc.

Reply:
I don't see a point of it, really... the whole function is basically a `switch` case with no extra logic.

Reply:
We need some test of this behavior. At minimum a test that when auto is set, X protocol version is used which could be a unit test (though yes kind of crappy). Optimally, some integration test that sets versions of client and server and makes sure the handshake fails. Or some other test that you cook up and think is better?

Reply:
Most of the tests for things like this, in the ContextConfig, end up going in connection_impl_test; I think a test there for negotiating the expected protocol versions (both the default/unset case, and the specified case) makes sense.

Reply:
Yeah, I'll definitely add "integration" tests in `connection_impl_test.cc` as stated elsewhere, once SNI is merged.

What I meant here is that I don't see a point of unit testing `ContextConfigImpl::tlsVersionFromProto()`.

Reply:
As long as you add other tests, yes agreed.